### PR TITLE
Flav/stable use event source

### PR DESCRIPTION
**Pull request by flvndvd**
## Description

We have observed a significant increase in Redis connections over the past few weeks. Despite multiple efforts in https://github.com/dust-tt/dust/pull/6187 and https://github.com/dust-tt/dust/pull/6202 to improve our Redis connection hygiene, we have been unable to reduce the load sufficiently. The math is straightforward: on the private API, the maximum number of Redis connections for the stream corresponds either to the number of conversations/messages created per minute or the number of active users.

After spending some hours 🥵, I discovered issues with how we handle the event stream at the conversation level. Specifically, the `useEventSource` hook is being re-created and re-opening a new connection on every render, resulting in unnecessary remounts and causing excess Redis connections. If you inspect the network, you will notice numerous calls to the `/events` endpoint.

This PR is based on the principle that our streaming logic should be independent of render cycles. Regardless of the number of re-renders, only one connection per conversationId/messageId should be maintained. This PR implements a stable, global EventSource manager to ensure consistent EventSource connections across renders and component lifecycles.

### Changes
- Created `stableEventSourceManager` as a global to manage EventSource instances.
- Modified the `useEventSource` hook to use the stable manager instead of creating a new one on each render.
- Optimized and restructured event handling.

## Testing

- [x] Verified that EventSource connections remain stable across component re-renders.
- [x] Checked that participant lists are only updated when necessary.
- [x] Ensured existing functionality remains intact.
- [x] Test on front-edge

## Risk

Worst case it breaks the conversation/message streaming logic. Thoroughly tested locally. Safe to rollback.

### CODE REVIEW

**Comment by spolu** [comment]
(ack will review tomorrow morning. tricky one :+1:)

**Review by PopDaph — APPROVED** [review]
As seen IRL I don't get everything, clearly I'm not familiar with this code but trusting you with it especially as you said you tested it thoroughly. Might want a review or PMRR from @lasryaric who wrote the initial logic?

File: front/hooks/useEventSource.ts [hunk 1]
```diff
@@ -1,67 +1,173 @@
...
+ * - Create a new EventSource: stableEventSourceManager.create(url, uniqueId)
+ * - Retrieve an existing EventSource: stableEventSourceManager.get(uniqueId)
+ * - Close and remove an EventSource: stableEventSourceManager.remove(uniqueId)
+ *
+ * This manager is designed to be used in conjunction with the useEventSource hook
+ * to provide stable and efficient EventSource handling across React component lifecycles.
+ */
+const stableEventSourceManager = {
+  // Map to store active EventSource instances, keyed by unique identifiers
+  sources: new Map<string, EventSource>(),
```
Comment (lasryaric) [inline front/hooks/useEventSource.ts:29]: Can we have a weakmap here? This will prevent leaking these connections if we forget to call manager.remove().

**Review by lasryaric — COMMENTED** [review]
Left one comment.

**Review by lasryaric — APPROVED** [review]
LGTM

**Comment by spolu** [comment]
LGTM :+1: Not fully understood but looks very reasonable \o/

**Pull request merged**